src/pdf_beamtime/src/inner_state_machine.cpp

```diff
@@ -0,0 +1,168 @@
+#include <pdf_beamtime/inner_state_machine.hpp>
+
+InnerStateMachine::InnerStateMachine(
+  const rclcpp::Node::SharedPtr node, State external_state_enum)
+: node_(node), external_state_enum_(external_state_enum)
+{
+  internal_state_enum_ = Internal_State::RESTING;
+
+}
+
+InnerStateMachine::~InnerStateMachine()
+{
+}
+
+void InnerStateMachine::set_joint_goal(std::vector<double> joint_goal)
+{
+  joint_goal_ = joint_goal;
+}
+
+State InnerStateMachine::move_robot(
+  moveit::planning_interface::MoveGroupInterface & mgi)
+{
+
+  if (internal_state_enum_ == Internal_State::RESTING) {
+    RCLCPP_INFO(
+      node_->get_logger(), "[%s] Robot's current internal state is %s ",
+      external_state_names_[static_cast<int>(external_state_enum_)].c_str(),
+      internal_state_names[static_cast<int>(internal_state_enum_)].c_str());
+
+    mgi.setJointValueTarget(joint_goal_);
+
+    // Create a plan to that target pose
+    auto const [success, plan] = [&mgi] {
```

Comment:
Can we refactor `success` here to be `planning_success`? I know it's a bit verbose, but would clarify with many state successes rolling around.

Reply:
Words success was changed to planning_success
https://github.com/maffettone/erobs/blob/c4b4a33b406f989bd7f26a6cec1318fabdf4f4d4/src/pdf_beamtime/src/inner_state_machine.cpp#L26